skimage/morphology/gray.py

```diff
@@ -222,19 +186,32 @@ def erosion(image, footprint=None, out=None, shift_x=False, shift_y=False):
     if out is None:
         out = np.empty_like(image)
 
+    if mode is None:
+        mode = "constant"
+        if np.issubdtype(out.dtype, bool):
+            cval = True
+        elif np.issubdtype(out.dtype, np.integer):
+            cval = np.iinfo(image.dtype).max
+        else:
+            cval = np.inf
```

Comment:
`ndi.grey_erosion` doesn't have this boundary extension mode. You could call it "max" and "min", but it's nice to have one boundary extension mode that means "max" in the erosion and "min" in the dilation. Because now we can use this mode in the opening and closing, and get the right behavior there. So I called it `None`. I would make this the default mode in skimage2, because it's the mode that makes most sense for all morphological operators, it never causes weird artefacts at the image boundary.

Of course we could suggest adding this mode to `ndimage`...

Reply:
Ah, that sound's useful indeed. Though I would prefer to be a bit more specific about the name. That way we can also preserve `None` in case we ever need it for a deprecation or something else. What do you think about the naming the modes "dtype_max" and "dtype_min" respectively?

I think it wouldn't hurt to ask SciPy if they are interested to add this mode. If they are interested we should probably agree on a common name for this mode. But waiting for SciPy shouldn't keep us from adding it in this PR in the meantime.

Reply:
I've added "max" and "min" modes, and an "ignore", which is interpreted as "max" by the erosion and as "min" by the dilation. This makes it possible to pass "ignore" to the opening or closing, and get the right behavior.